src/collection.ts

```diff
@@ -210,7 +218,7 @@ class Collection {
    * @memberof Collection#
    * @readonly
    */
-  get collectionName() {
+  get collectionName(): string {
```

Comment:
This is something we're going to encounter along our full transition to TS but if namespace.collection was correctly typed as a string then the return type annotation is not needed. But, understandably, we don't want the changes here to sprawl just to put type annotations in their ideal location. I took a quick look and it looks like as of yet there's not a linting rule to help us here, but something we can keep in mind / discuss best path

Reply:
I think we should discuss this more at length with the whole team, but I think I'm currently of a mind that if we're adding type annotations, we should always annotate explicitly. Definitely in this case I was trying to just force the type somewhere, but in general I think we should have a rule